integration/spring-data/base/src/main/java/com/blazebit/persistence/spring/data/base/query/AbstractPartTreeBlazePersistenceQuery.java

```diff
@@ -76,8 +78,9 @@ public AbstractPartTreeBlazePersistenceQuery(EntityViewAwareJpaQueryMethod metho
         this.tree = new PartTree(method.getName(), domainClass);
         this.parameters = method.getJpaParameters();
 
-        this.query = isCountProjection(tree) ? new AbstractPartTreeBlazePersistenceQuery.CountQueryPreparer(persistenceProvider, parameters.potentiallySortsDynamically())
-                : new AbstractPartTreeBlazePersistenceQuery.QueryPreparer(persistenceProvider, parameters.potentiallySortsDynamically());
+        boolean recreateQueries = parameters.potentiallySortsDynamically();
```

Comment:
This should be `boolean recreateQueries = parameters.potentiallySortsDynamically() || parameters.hasOptionalParameters();` no? Otherwise it would be cached and a second invocation wouldn't add the optional parameters.